trulens/visualizations.py

```diff
@@ -999,3 +1012,301 @@ def __call__(
             attrs_input, x, output_file, blur, threshold, masked_opacity,
             combine_channels
         )
+
+
+class Output(ABC):
+    """Base class for visualization output formats."""
+    
+    @abstractmethod
+    def blank(self) -> str:
+        ...
+
+    @abstractmethod
+    def space(self) -> str:
+        ...
+
+    @abstractmethod
+    def escape(self, s: str) -> str:
+        ...
+
+    @abstractmethod
+    def line(self, s: str) -> str:
+        ...
+
+    @abstractmethod
+    def magnitude_colored(self, s: str, mag: float) -> str:
+        ...
+
+    @abstractmethod
+    def append(self, *parts: Iterable[str]) -> str:
+        ...
+
+    @abstractmethod
+    def render(self, s: str) -> str:
+        ...
+
+
+# TODO(piotrm): implement a latex output format
+
+
+class Text(Output):
```

Comment:
nit: if we are saying Output is TextOutput, maybe we call this PlainText

Reply:
Renamed to PlainText .